src/ring/middleware/session_timeout.clj

```diff
@@ -23,17 +44,32 @@
          (if (map? timeout-response)
            (nil? timeout-handler)
            (ifn? timeout-handler))]}
-  (fn [request]
-    (let [session  (:session request {})
-          end-time (::idle-timeout session)]
-      (if (and end-time (< end-time (current-time)))
-        (assoc (or timeout-response (timeout-handler request)) :session nil)
-        (when-let [response (handler request)]
-          (let [session (:session response session)]
-            (if (nil? session)
-              response
-              (let [end-time (+ (current-time) timeout)]
-                (assoc response :session (assoc session ::idle-timeout end-time))))))))))
+  (fn
+    ([request]
+     (if (session-idle-expired? request)
+       (assoc (or timeout-response (timeout-handler request)) :session nil)
+       (let [response (handler request)
+             session  (:session response (:session request {}))]
```

Comment:
This can be moved into `idle-session-timeout-response`. See my [example code](https://github.com/ring-clojure/ring-session-timeout/pull/7#issuecomment-1463087485) for what I mean. See how the request session is pulled out at the top, and then passed into the response generator? This ensures we don't need to pass the whole request around when only the session is necessary.